src/components/Composer/index.js

```diff
@@ -197,46 +207,73 @@ class Composer extends React.Component {
     }
 
     /**
-     * Handles all types of drag-N-drop events on the composer
+     * DragEvent
      *
      * @param {Object} e native Event
      */
-    dragNDropListener(e) {
-        let isOriginComposer = false;
-        const handler = () => {
-            // Setting dropEffect for dragover is required for '+' icon on certain platforms/browsers (eg. Safari)
-            switch (e.type) {
-                case 'dragover':
-                    e.preventDefault();
-                    e.dataTransfer.dropEffect = COPY_DROP_EFFECT;
-                    this.props.onDragOver(e, isOriginComposer);
-                    break;
-                case 'dragenter':
+    dropZoneDragHandler = (e) => {
+        // Setting dropEffect for dragover is required for '+' icon on certain platforms/browsers (eg. Safari)
+        switch (e.type) {
+            case 'dragover':
+                // Continuous event -> can hurt performance, be careful when subscribing
+                e.dataTransfer.dropEffect = COPY_DROP_EFFECT;
+                this.dropZoneDragState = 'dragover';
+                this.props.onDragOver(e);
+                break;
+            case 'dragenter':
+                // Avoid reporting onDragEnter for children views -> not performant
```

Comment:
By using dropZoneDragState which is last detected drag state on the dropzone -> we start with dragleave since user is not dragging initially.
 This state is updated when drop zone is left/entered entirely(not taking the children in the account) or entire window is left

Reply:
Sorry, I meant in the comment. :smile:

Reply:
I did that as well :)